lib/providers.dart

```diff
@@ -0,0 +1,74 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'todo_item.dart';
+
+final todoAppProvider =
+    ChangeNotifierProvider<AppChangeNotifier>((ref) => AppChangeNotifier());
+
+class AppChangeNotifier extends ChangeNotifier {
+  final List<TodoItem> unCompletedList = [
+    TodoItem(
+      id: 0,
+      title: 'dart',
+      content: 'dart-content',
+      isCompleted: false,
+    ),
+    TodoItem(
+      id: 1,
+      title: 'ssh',
+      content: 'ssh',
+      isCompleted: false,
+    ),
+  ];
+  final List<TodoItem> completedList = [
+    TodoItem(
+      id: 2,
+      title: 'python',
+      content: '再帰関数',
+      isCompleted: true,
+    ),
+    TodoItem(
+      id: 3,
+      title: 'flutter',
+      content: '非同期処理',
+      isCompleted: true,
+    ),
+  ];
+
+  int bottomIndex = 0;
+
+  void bottomIndexChange(int index) {
+    bottomIndex = index;
+    notifyListeners();
+  }
+
+  void addTodoItem(Map<String, String> formValue) {
+    unCompletedList.add(
+      TodoItem(
+        id: unCompletedList.length + completedList.length,
+        title: formValue['title']!,
```

Comment:
`!` オペレーターはエディター上での `null` エラーを消すためのものなので、エラーの原因に気づけなかったり予想外のバグが起きたりするのでできるだけ控えた方がいいです。特に `Map` から取り出しは自由度が高すぎるが故にエラーがおきやすいので、定義したクラスのインスタンスなんかを使うとより安全になります。
```suggestion
        title: formValue['title'] ?? '',
```